crates/wasi-http/src/lib.rs

```diff
@@ -95,29 +95,29 @@ impl TryFrom<http::Request<Bytes>> for Request {
     }
 }
 
-pub struct Method<'a>(http_types::MethodParam<'a>);
+pub struct Method(http_types::Method);
 
 #[cfg(feature = "std")]
-impl<'a> Deref for Method<'a> {
-    type Target = http_types::MethodParam<'a>;
+impl Deref for Method {
+    type Target = http_types::Method;
 
     fn deref(&self) -> &Self::Target {
         &self.0
     }
 }
 
-impl<'a> From<http::Method> for Method<'a> {
+impl From<http::Method> for Method {
     fn from(method: http::Method) -> Self {
         Self(match method {
-             http::Method::GET => http_types::MethodParam::Get,
-             http::Method::POST => http_types::MethodParam::Post,
-             http::Method::PUT => http_types::MethodParam::Put,
-             http::Method::DELETE => http_types::MethodParam::Delete,
-             http::Method::PATCH => http_types::MethodParam::Patch,
-             http::Method::CONNECT => http_types::MethodParam::Connect,
-             http::Method::TRACE => http_types::MethodParam::Trace,
-             http::Method::HEAD => http_types::MethodParam::Head,
-             http::Method::OPTIONS => http_types::MethodParam::Options,
+             http::Method::GET => http_types::Method::Get,
+             http::Method::POST => http_types::Method::Post,
+             http::Method::PUT => http_types::Method::Put,
+             http::Method::DELETE => http_types::Method::Delete,
+             http::Method::PATCH => http_types::Method::Patch,
+             http::Method::CONNECT => http_types::Method::Connect,
+             http::Method::TRACE => http_types::Method::Trace,
+             http::Method::HEAD => http_types::Method::Head,
+             http::Method::OPTIONS => http_types::Method::Options,
              _ => panic!("failed due to unsupported method, currently supported methods are: GET, POST, PUT, DELETE, PATCH, CONNECT, TRACE, HEAD, and OPTIONS"),
```

Comment:
`http_types::Method` has a `FromStr` method we could use to try parsing the `http::Method`. I don't think we should have panics in here, and instead use `TryFrom`.

Reply:
with the wit-bindgen v0.4, i don't have the `FromStr` nor the `http_types::Method`

Reply:
i am planning on updating those dependencies when we have the next major bump in wasmtime

Reply:
i implemented the tryfrom and returned an error instead of panicing